firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataConverter.java

```diff
@@ -194,25 +192,30 @@ public FieldValue parseQueryValue(Object input) {
 
   /** Converts a POJO to native types and then parses it into model types. */
   private FieldValue convertAndParseData(Object input, ParseContext context) {
-    UserData.Source source = context.getDataSource();
+    Object converted = CustomClassMapper.convertToPlainJavaTypes(input);
+    return parseData(converted, context);
+  }
+
+  /**
+   * Wrapper around convertAndParseData() that expects input to conform to document data (in
```

Comment:
If you rename the above method, I would not mention the implementation detail in the comment (or just inline the two line call).

Reply:
done